modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/MailActivityBehavior.java

```diff
@@ -98,7 +102,7 @@ public void execute(CommandContext commandContext, PlanItemInstanceEntity planIt
             getFilesFromFields(attachments, planItemInstanceEntity, files, dataSources);
 
             if (StringUtils.isAllEmpty(toStr, ccStr, bccStr)) {
-                throw new FlowableException("No recipient could be found for sending email");
+                throw new FlowableException("No recipient could be found for sending email: fields to ("+ getEmailExprText(to)+"), cc ("+ getEmailExprText(cc)+") and bcc ("+ getEmailExprText(bcc)+") are all resolved to an empty value");
```

Comment:
It appears this will throw a NPE if for example the `to` address is missing.  The `expression` is tested for null in `getEmailExprText` but not the value it contains.